packages/twenty-server/src/engine/workspace-manager/workspace-sync-metadata/services/workspace-sync-field-metadata.service.ts

```diff
@@ -137,4 +108,120 @@ export class WorkspaceSyncFieldMetadataService {
+        workspaceFeatureFlagsMap,
+      );
+
+    // Create map of original and standard object metadata by standard ids
+    const originalObjectMetadataMap = mapObjectMetadataByUniqueIdentifier(
+      originalObjectMetadataCollection,
+    );
+
+    this.logger.log('Comparing standard objects and fields metadata');
+
+    // Loop over all standard objects and compare them with the objects in DB
+    for (const [
+      standardObjectId,
+      standardFieldMetadataCollection,
+    ] of standardObjectStandardFieldMetadataMap) {
+      const originalObjectMetadata =
+        originalObjectMetadataMap[standardObjectId];
+      const computedStandardFieldMetadataCollection = computeStandardFields(
+        standardFieldMetadataCollection,
+        originalObjectMetadata,
+        // We need to provide this for generated relations with custom objects
+        customObjectMetadataCollection,
+      );
+
+      const fieldComparatorResults = this.workspaceFieldComparator.compare(
+        originalObjectMetadata.id,
+        originalObjectMetadata.fields,
+        computedStandardFieldMetadataCollection,
+      );
+
+      for (const fieldComparatorResult of fieldComparatorResults) {
+        switch (fieldComparatorResult.action) {
+          case ComparatorAction.CREATE: {
+            storage.addCreateFieldMetadata(fieldComparatorResult.object);
+            break;
+          }
+          case ComparatorAction.UPDATE: {
+            storage.addUpdateFieldMetadata(fieldComparatorResult.object);
+            break;
+          }
+          case ComparatorAction.DELETE: {
+            storage.addDeleteFieldMetadata(fieldComparatorResult.object);
+            break;
+          }
+        }
+      }
+    }
+  }
+
+  private async synchronizeCustomObjectFields(
+    context: WorkspaceSyncContext,
+    customObjectMetadataCollection: ObjectMetadataEntity[],
+    storage: WorkspaceSyncStorage,
+    workspaceFeatureFlagsMap: FeatureFlagMap,
+  ): Promise<void> {
+    // Create standard field metadata collection
+    const customObjectStandardFieldMetadataCollection =
+      this.standardFieldFactory.create(
+        CustomWorkspaceEntity,
+        context,
+        workspaceFeatureFlagsMap,
+      );
+
+    // Loop over all custom objects from the DB and compare their fields with standard fields
+    for (const customObjectMetadata of customObjectMetadataCollection) {
+      // Also, maybe it's better to refactor a bit and move generation part into a separate module ?
+      const standardFieldMetadataCollection = computeStandardFields(
+        customObjectStandardFieldMetadataCollection,
+        customObjectMetadata,
+      );
+
+      /**
+       * COMPARE FIELD METADATA
+       */
+      const fieldComparatorResults = this.workspaceFieldComparator.compare(
+        customObjectMetadata.id,
+        customObjectMetadata.fields,
+        standardFieldMetadataCollection,
+      );
+
+      for (const fieldComparatorResult of fieldComparatorResults) {
```

Comment:
this part seems to be similar to the other function here, maybe it can be extracted to another function?